### PR TITLE
Make argsAlloc/ArgIterator return zero-sentinel strings

### PR DESCRIPTION
**Pull request by chwayne**
Implements #6657

### CODE REVIEW

File: lib/std/process.zig [hunk 1]
```diff
@@ -248,7 +248,7 @@ pub const ArgIteratorWasi = struct {
             else => |err| return os.unexpectedErrno(err),
         }
 
-        var result_args = try allocator.alloc([]u8, count);
+        var result_args = try allocator.alloc([:0]u8, count);
```
Comment (daurnimator) [inline lib/std/process.zig:251]: Is this valid? don't you need `.allocSentinel` ?

File: lib/std/process.zig [hunk 2]
```diff
@@ -515,35 +515,36 @@ pub fn argsAlloc(allocator: *mem.Allocator) ![][]u8 {
     while (it.next(allocator)) |arg_or_err| {
         const arg = try arg_or_err;
         defer allocator.free(arg);
-        try contents.appendSlice(arg);
+        try contents.appendSlice(@ptrCast([*]u8, arg.ptr)[0 .. arg.len + 1]);
```
Comment (daurnimator) [inline lib/std/process.zig:518]: Why is this cast required?

File: lib/std/process.zig (hunk 1 above)
Comment (chwayne) [reply lib/std/process.zig:251]: No, the type of the return value is ```[][:0]u8```, which is correct.

File: lib/std/process.zig (hunk 2 above)
Comment (chwayne) [reply lib/std/process.zig:518]: Oh it's not. I assumed that trying to slice one past the end of the slice would trigger bounds-checking, so I put the cast there. But just realized it's not needed. Will fix.

**Comment by andrewrk** [comment]
Nice work!

**Pull request merged**